net/bridge.go

```diff
@@ -457,14 +458,27 @@ func configureIPTables(config *BridgeConfig, ips ipset.Interface) error {
 	}
 
 	if config.NPC {
-		// Steer traffic via the NPC
-		_ = ipt.NewChain("filter", "WEAVE-NPC") // ignore error because it might already exist
-		// If WEAVE-NPC chain doesn't exist then creating a rule in the chain will fail
+		// Steer traffic via the NPC.
+
+		// Ignore error because chains might already exist
+		_ = ipt.NewChain("filter", npc.MainChain)
+		_ = ipt.NewChain("filter", npc.EgressChain)
```

Comment:
May be check if chain exists by `ListChains` of the `filter` table? Just ignoring the error might mask any real error.

Reply:
Agreed!

Reply:
I've introduced `ensureChains` function which does what you suggest.